reflection.md

```diff
@@ -2,17 +2,51 @@
 
 ## Which operators were easiest to implement and why?
 
+I thought the easiest operators to implement were || and ~. The method for implementing these was very similar to the example with complex numbers in class. There is no recursion to consider, and the parameters for Union and Concat seemed pretty obvious. I've also made datatypes in previous assignments in Haskell and Java, so this wasn't hard to figure out. 
+
 ## Which operators were most difficult to implement and why?
 
+Something that I struggled with was getting  " val cThree = 'c'{3} " to work. The error message kept displaying that Char('c') did not take parameters, even though I wrote a method, "  implicit def Char2Lit (c : Char) = new Literal (c); " to convert from character to literal. 
+
+It also took me a while to understand " object RegularExpression " versus "abstract class RegularExpression ", and the relationship between the two. These are concepts that I've never encountered in Java before.
+
 ## Comment on the design of this internal DSL
 
 Write a few brief paragraphs that discuss:
    + What works about this design? (For example, what things seem easy and
    natural to say, using the DSL?)
+   
+   It's much more convenient to chain literals together in a union. Under the previous language, the user had to specify the exact "associative order" of what pairs of literals are put into a union first, before which other pairs of literals are added on as unions. It's much simpler just to say '0' || '1' || '2' || '3' || '4' || '5' || '6' || '7' || '8' || '9' and not have to specify the order that characters are chained together in an expression that expresses a union of elements. 
```

Comment:
While I agree the newer syntax is easier, I think chaining syntax like

```scala
Literal('0').Union().Literal('1').Union().Literal('2').Union().Literal('3').Union().Literal('4').Union().Literal('5').Union().Literal('6').Union().Literal('7').Union().Literal('8').Union().Literal('9').Union()
```

would be better than the compositional/nesting of the original

```scala
Union(Literal('0'), Union(Literal('1'), Union(...)))
```

Either way, the `||` DSL-y syntax is a lot nicer to type and look at as you mention. Scala's leniency with omitting parentheses, using operators as infixes, and naming them with non alpha characters makes this DSL work really well.